lib/src/main/java/com/knuddels/jtokkit/TokenEncoderLarge.java

```diff
@@ -33,10 +34,12 @@ static int calculateTokensLarge(TokenEncoder tokenEncoder, int maxTokenCount, bo
         }
 
         while (validRanks > 0) {
-            TreeMap<Integer, RankNode> minNodes = rankMap.firstEntry().getValue();
-            for (int i = 0; i < minNodes.size(); i++) {
-                RankNode minNode = minNodes.firstEntry().getValue();
-                assert minNode.rank != MAX_RANK;
+            TreeMap<Integer, RankNode> minNodes = rankMap.pollFirstEntry().getValue();
+            int firstIndex;
+            for (Entry<Integer, RankNode> entry = minNodes.firstEntry(); entry != null; entry = minNodes.ceilingEntry(firstIndex)) {
```

Comment:
because of the tree structure we're actually storing every instance of the same token, so once we find any of them, we can be sure that the next few minimums will also be the same token - so let's just iterate those instead, without removing them one-by-one.